### DISCUSSION TITLE
how does FAR determines resize?

### DISCUSSION THREAD

**Discussion by determin1st** [FarGroup/FarManager#829] [General]
for example im running FAR in ConEmu which window allows to resize its height and width, detecting of the width change is quite simple - it is reported by `ReadConsoleInput` windows api function giving `WINDOW_BUFFER_SIZE_EVENT`. but the height change is not reported that way.

so i was thinking how me, the terminal user should fetch that change. is the FAR uses some ticker-poller or is it there some magic ESC sequence to subscribe to resize?

**Comment by yegor-mialyk** [comment]
```
typedef struct _WINDOW_BUFFER_SIZE_RECORD {
  COORD dwSize;
} WINDOW_BUFFER_SIZE_RECORD;
```

`dwSize` is the `COORD` structure containing `X` and `Y`

Sample: https://learn.microsoft.com/en-us/windows/console/reading-input-buffer-events

**Reply by determin1st** [reply]
those i know, but when you resize ConEmu window vertically, it doesnt strike, although FAR changes / redraws itself. there must be another way of detecting window change.

also, the Y is useless in that structure, its always 32766, number of lines that buffer stores

**Reply by determin1st** [reply]
![lasjf - копия](https://github.com/FarGroup/FarManager/assets/16524081/23f27e14-a6b3-4ec3-8fc8-8003489acd6d)

**Reply by determin1st** [reply]
https://github.com/microsoft/terminal/discussions/14975

**Comment by yegor-mialyk** [comment]
If you need a size of visible (display) part of the window, you can call `GetConsoleScreenBufferInfo()`.
